metaseq/distributed/utils.py

```diff
@@ -179,14 +180,18 @@ def distributed_init(cfg: MetaseqConfig):
 
 
 def distributed_main(i, main, cfg: MetaseqConfig, kwargs):
+    if not cfg.distributed_training.distributed_no_spawn:
```

Comment:
Can we remove `distributed_no_spawn` config entirely? Will we have `ntasks_per_node` > 1?

Reply:
No, DNS is great and useful.

Reply:
DNS is particularly useful during our training runs or our API runs, as it let's any exceptions in the subworkers get noticed by slurm and kills (or requeues) the entire job. If you use spawning, then there's a good chance an exception anywhere OUTSIDE of the local rank 0 will result in a hang.

Reply:
When is DNS == False useful then?

Reply:
Local command line, because now you don't get the luxury of SLURM handling launching.